atom/renderer/lib/override.coffee

```diff
@@ -118,6 +118,12 @@ Object.defineProperty window.history, 'length',
   get: ->
     getHistoryOperation 'length'
 
-# Make document.hidden return the correct value.
+# Make document.hidden and document.visibilityState return the correct value.
 Object.defineProperty document, 'hidden',
-  get: -> !remote.getCurrentWindow().isVisible()
+  get: ->
+    currentWindow = remote.getCurrentWindow()
+    !currentWindow.isFocused() || !currentWindow.isVisible()
```

Comment:
Window without focus can also be visible, we should check `isMinimized` instead of `isFocused`.